### ISSUE TITLE
Cannot run on ASF 6.0.0.3

### ISSUE THREAD

**Issue by Cicini** [Citrinate/BoosterManager#13]
It seems that this plugin is not working properly after the ASF update
```
2024-03-09 16:19:14|ArchiSteamFarm-2459969|ERROR|ASF|InitPlugins() Failed due to error: BoosterManager, Version=2.8.1.0, Culture=neutral, PublicKeyT>2024-03-09 16:19:14|ArchiSteamFarm-2459969|ERROR|ASF|InitPlugins() System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the >Method 'OnASFInit' in type 'BoosterManager.BoosterManager' from assembly 'BoosterManager, Version=2.8.1.0, Culture=neutral, PublicKeyToken=null' doe>Method 'Save' in type 'BoosterManager.BoosterDatabase' from assembly 'BoosterManager, Version=2.8.1.0, Culture=neutral, PublicKeyToken=null' does no>Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>
Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the fi>

   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at ArchiSteamFarm.Plugins.PluginsCore.InitPlugins()
System.TypeLoadException: Method 'OnASFInit' in type 'BoosterManager.BoosterManager' from assembly 'BoosterManager, Version=2.8.1.0, Culture=neutral>System.TypeLoadException: Method 'Save' in type 'BoosterManager.BoosterDatabase' from assembly 'BoosterManager, Version=2.8.1.0, Culture=neutral, Pu>System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aee>
```

**Comment by Citrinate** [comment]
Just released V2.8.3 of the plugin which works with ASF 6.0.0.3

**Comment by Cicini** [comment]
> Just released V2.8.3 of the plugin which works with ASF 6.0.0.3
It seems that there are still related errors, I am not sure if it will affect the plugin's functionality
```
2024-03-09 16:35:42|ArchiSteamFarm-2460271|ERROR|ASF|Community with ASFEnhance failed() System.IO.FileNotFoundException: Could not load file or assembly 'ASFEnhance, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'ASFEnhance, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.<InternalLoad>g____PInvoke|49_0(NativeAssemblyNameParts*, ObjectHandleOnStack, StackCrawlMarkHandle, Int32, ObjectHandleOnStack, ObjectHandleOnStack)
   at System.Reflection.RuntimeAssembly.InternalLoad(NativeAssemblyNameParts*, ObjectHandleOnStack, StackCrawlMarkHandle, Boolean , ObjectHandleOnStack, ObjectHandleOnStack)
   at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName, StackCrawlMark&, AssemblyLoadContext , RuntimeAssembly , Boolean )
   at System.Reflection.RuntimeAssembly.InternalLoad(String, StackCrawlMark&, AssemblyLoadContext )
   at System.Reflection.Assembly.Load(String)
   at BoosterManager.AdapterBridge.InitAdapter(String pluginName, String pluginId, String cmdPrefix, String repoName, MethodInfo cmdHandler)
```

**Comment by Cicini** [comment]
> > Just released V2.8.3 of the plugin which works with ASF 6.0.0.3
> 
> It seems that there are still related errors, I am not sure if it will affect the plugin's functionality
> 
> ```
> 2024-03-09 16:35:42|ArchiSteamFarm-2460271|ERROR|ASF|Community with ASFEnhance failed() System.IO.FileNotFoundException: Could not load file or assembly 'ASFEnhance, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
> 
> File name: 'ASFEnhance, Culture=neutral, PublicKeyToken=null'
>    at System.Reflection.RuntimeAssembly.<InternalLoad>g____PInvoke|49_0(NativeAssemblyNameParts*, ObjectHandleOnStack, StackCrawlMarkHandle, Int32, ObjectHandleOnStack, ObjectHandleOnStack)
>    at System.Reflection.RuntimeAssembly.InternalLoad(NativeAssemblyNameParts*, ObjectHandleOnStack, StackCrawlMarkHandle, Boolean , ObjectHandleOnStack, ObjectHandleOnStack)
>    at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName, StackCrawlMark&, AssemblyLoadContext , RuntimeAssembly , Boolean )
>    at System.Reflection.RuntimeAssembly.InternalLoad(String, StackCrawlMark&, AssemblyLoadContext )
>    at System.Reflection.Assembly.Load(String)
>    at BoosterManager.AdapterBridge.InitAdapter(String pluginName, String pluginId, String cmdPrefix, String repoName, MethodInfo cmdHandler)
> ```

After logging in

```
2024-03-09 16:40:54|ArchiSteamFarm-2460470|ERROR|ASF|GetBoosterPage() System.MissingMethodException: Method not found: 'Void System.Text.Json.JsonSerializerOptions.set_NumberHandling(System.Text.Json.Serialization.JsonNumberHandling)'.
   at BoosterManager.BoosterPageResponse..ctor(Bot bot, IDocument boosterPage)
   at BoosterManager.WebRequest.GetBoosterPage(Bot bot)
2024-03-09 16:40:54|ArchiSteamFarm-2460470|ERROR|Cicini|UpdateBoosterInfos() boosterPage is null!
2024-03-09 16:40:54|ArchiSteamFarm-2460470|ERROR|Cicini|Run() Failed to update booster information
```

**Comment by Citrinate** [comment]
The ASFEnhance error should not be an issue.  Are you using something other than ASF-generic?

**Comment by Cicini** [comment]
> The ASFEnhance error should not be an issue. Are you using something other than ASF-generic?

I still get the same error after I have cleared all files and redownloaded and configured them （Only BoosterManager Installed)

config:

```
{
  "Enabled": true,
  "GamesPlayedWhileIdle": [
    2230650
  ],
  "OnlineStatus": 7,
  "SteamLogin": "********",
  "SteamPassword": "********",
  "GamesToBooster": [
    1406990
  ]
}
```

**Comment by Citrinate** [comment]
The `System.MissingMethodException` error to me suggests you're using something other than ASF-generic, maybe ASF-win-x64 if you're using Windows?

This newest version has only been tested with ASF-generic, and so it's very possible to be broken with other variants of ASF.  If that's the case, then I don't have a fix for this right now, but will look into addressing this within a few days.

**Comment by Cicini** [comment]
> The `System.MissingMethodException` error to me suggests you're using something other than ASF-generic, maybe ASF-win-x64 if you're using Windows?
> 
> This newest version has only been tested with ASF-generic, and so it's very possible to be broken with other variants of ASF. If that's the case, then I don't have a fix for this right now, but will look into addressing this within a few days.

I'm using ASF-linux-x64

**Comment by phrostb** [comment]
I'm using asf-win-x64 and getting the same error. i don't mind waiting for a fix, just wanted to say thanks for keeping this project going! i appreciate you!

**Comment by JustArchi** [comment]
> The `System.MissingMethodException` error to me suggests you're using something other than ASF-generic, maybe ASF-win-x64 if you're using Windows?
> 
> This newest version has only been tested with ASF-generic, and so it's very possible to be broken with other variants of ASF. If that's the case, then I don't have a fix for this right now, but will look into addressing this within a few days.

That specific issue reported above you can fix by using `[JsonNumberHandling]` attribute on class properties rather than json serializer (global) setting :slightly_smiling_face:

**Comment by Citrinate** [comment]
New release is out which fixes this, [V2.8.4](https://github.com/Citrinate/BoosterManager/releases/tag/2.8.4).  It's possible there's other errors I didn't find though, but I'm still closing this as the original problem this issue was opened for is certainly fixed.  Feel free to open a new one if there's any other problems.

**Comment by CuteistFox** [comment]
just wondering incase of more issues, can you just copy the config and plugins directorys from a linux-x64 install to a generic one?
